server/src/main/java/io/druid/timeline/partition/SingleDimensionShardSpec.java

```diff
@@ -112,6 +118,24 @@ public ShardSpec getShardSpec(long timestamp, InputRow row)
     };
   }
 
+  @Override
+  public Map<String, RangeSet<String>> getDomain()
+  {
+    Map<String, RangeSet<String>> retMap = new HashMap<>();
+    Range<String> range;
+    if (start == null && end == null) {
+      range = Range.all();
+    } else if (start == null) {
+      range = Range.atMost(end);
+    } else if (end == null) {
+      range = Range.atLeast(start);
+    } else {
+      range = Range.closed(start, end);
+    }
+    retMap.put(dimension, ImmutableRangeSet.of(range));
```

Comment:
`ImmutableMap.of(dimension, ImmutableRangeSet.of(range))` is a bit more efficient (immutablemap has an optimization for zero and one element maps)